numpy/lib/stride_tricks.py

```diff
@@ -111,6 +111,92 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
 
     return view
 
+def sliding_window_view(x, shape, step=None):
+    """
+    Create rolling window views of the 2D array with the given shape.
+
+    For some cases, there may be more efficient approaches
+
+    Parameters
+    ----------
+    x : ndarray
+        Array to create rolling window views.
+    shape : sequence of int
+        The shape of the sliding window.
+    step: sequence of int, optional
+        The sliding step of view window for each dimension of input. Defaults to 1 on all dimensions.
+
+    Returns
+    -------
+    view : ndarray
+
+    Examples
+    --------
+    >>> x = np.arange(12).reshape(3,4)
+    >>> shape = [2,2]
+    >>> np.lib.stride_tricks.sliding_window_view(x, shape)
+    array([[[[ 0,  1],
+         [ 4,  5]],
+
+        [[ 1,  2],
+         [ 5,  6]],
+
+        [[ 2,  3],
+         [ 6,  7]]],
+
+
+       [[[ 4,  5],
+         [ 8,  9]],
+
+        [[ 5,  6],
+         [ 9, 10]],
+
+        [[ 6,  7],
+         [10, 11]]]])
+
+
+    >>> x = np.arange(12).reshape(3,4)
+    >>> shape = [2,2]
+    >>> step = [1,2]
+    >>> sliding_window_view(x, shape, step)
+    array([[[[ 0,  1],
+         [ 4,  5]],
+
+        [[ 2,  3],
+         [ 6,  7]]],
+
+
+       [[[ 4,  5],
+         [ 8,  9]],
+
+        [[ 6,  7],
+         [10, 11]]]])
```

Comment:
Missing 4 spaces at the beginning of every line here too

Reply:
fixed